### PR TITLE
[R2][Fixes] Added some fixes for logs and GUI

### PR DESCRIPTION
**Pull request by royjan**
**fixes:**

- item[key] == "true" -> item[key].lower() == "true"
- matplotlib won't show its debug details, only errors
- set severity from debug to info (better outputs)

**features:**

- thread_manager writes the best model to log
- model names change in graph (you can see in the presentation an example)
- a nice gif animation in about window
- an explain for RUN MODELS
- better chekbox labels to show parameters
- show errors popup

**refactor:**

- model_supports -> return_model_by_name
- ThreadManagerGUI -> cls
- filename='../history.log' -> filename='history.log'

### CODE REVIEW

File: main-gui.py [hunk 1]
```diff
@@ -195,36 +223,41 @@ def create_input_subwindow(self):
...
         if not best_model:
-            self.show_err_msg('You have to run train first!')
+            self.show_err_msg('You have to run train first!', user_input)
             return
         if not user_input:
-            self.show_err_msg("Input can't be empty!")
+            self.show_err_msg("Input can't be empty!", user_input)
             return
-        best_model.predict([[123, 24]])
+        # clf_result = best_model.predict([[123, 24]]) #TODO: let's talk about how we create a predict row
```
Comment (RickyDa) [inline main-gui.py:257]: We should!

File: main-gui.py [hunk 2]
```diff
@@ -48,14 +49,20 @@ def start_worker(self):
         Worker starts in a separated thread, doing its work and shows the results
         """
         global best_model
-        self.reset_check_boxes.emit()
-        ThreadManagerGUI.running_threads_args(X_train, y_train, X_test, y_test, params)
-        ThreadManagerGUI.wait_for_all_threads(check_checkbox=self.check_checkbox)
-        best_model = ThreadManagerGUI.return_best_model()
-        self.show_graph.emit()
+        try:
```
Comment (RickyDanipog) [inline main-gui.py:52]: nice

File: main-gui.py (hunk 1 above)
Comment (RickyDanipog) [inline main-gui.py:257]: We should!

**Review by RickyDanipog — APPROVED** [review]
Gj malik

**Pull request merged**